ngraph/frontend/onnx_import/src/utils/recurrent.cpp

```diff
@@ -46,21 +46,22 @@ namespace ngraph
                 const auto x_pshape = m_map[OpInput::X].get_partial_shape();
                 const auto w_pshape = m_map[OpInput::W].get_partial_shape();
                 const auto r_pshape = m_map[OpInput::R].get_partial_shape();
-                NGRAPH_CHECK(x_pshape.rank().is_static() && x_pshape[0].is_static() &&
-                                 x_pshape[1].is_static(),
-                             "RecurrentSequence input X must have static \"seq_length\" and "
-                             "\"batch_size\" dimensions.");
-                NGRAPH_CHECK(w_pshape.rank().is_static() && w_pshape[0].is_static(),
-                             "RecurrentSequence input W must have static \"num_directions\" "
-                             "(outermost) dimension.");
-                NGRAPH_CHECK(r_pshape.rank().is_static() && r_pshape[2].is_static(),
-                             "RecurrentSequence input R must have static \"hidden_size\" "
-                             "(innermost) dimension.");
 
-                const std::size_t hidden_size = m_map[OpInput::R].get_shape().back();
-                const std::size_t batch_size = m_map[OpInput::X].get_shape().at(0);
-                const std::size_t num_directions = m_map[OpInput::W].get_shape().front();
+                // Get dimensions needed for default inputs creation
+                auto shape_of_x = std::make_shared<default_opset::ShapeOf>(m_map[OpInput::X]);
+                auto axes = default_opset::Constant::create(element::i32, Shape{1}, {0});
+                auto batch_size_node = std::make_shared<default_opset::Gather>(
+                    shape_of_x, default_opset::Constant::create(element::i32, Shape{1}, {0}), axes);
+                auto seq_length_node = std::make_shared<default_opset::Gather>(
+                    shape_of_x, default_opset::Constant::create(element::i32, Shape{1}, {1}), axes);
 
+                auto shape_of_r = std::make_shared<default_opset::ShapeOf>(m_map[OpInput::R]);
+                auto num_directions_node = std::make_shared<default_opset::Gather>(
+                    shape_of_r, default_opset::Constant::create(element::i32, Shape{1}, {0}), axes);
+                auto hidden_size_node = std::make_shared<default_opset::Gather>(
+                    shape_of_r, default_opset::Constant::create(element::i32, Shape{1}, {2}), axes);
```

Comment:
Please consider creating a helper function to avoid duplicating code

Reply:
This is a standard way of creating ops and from my point of view it's more clear to read code like this than introduce `builder` function for such a simple case.